Python/Product/PythonTools/PythonTools/Project/PytestConfigureInfoBar.cs

```diff
@@ -0,0 +1,222 @@
+﻿using Microsoft.PythonTools.Infrastructure;
+using Microsoft.PythonTools.Logging;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.PythonTools.Interpreter;
+using Task = System.Threading.Tasks.Task;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Documents;
+using Microsoft.PythonTools.Project;
+
+namespace Microsoft.PythonTools.Project {
+    /*
+     * Bugs: 
+     *      If an info bar is shown and then a user selects pytest from project settings->Tests->pytest, then the info bar will remain
+     *      If the user adds a config file after the project is loaded, the info bar will not be displayed
+     */
+    internal abstract class ConfigurePytestInfoBar : PythonInfoBar {
+        private readonly IInterpreterOptionsService _interpreterOptionsService;
+        private IPythonInterpreterFactory _interpreterFactory;
+        private bool IsGloballySuppressed = false; //TODO
+        private string InfoBarContext { get; set; }
+
+        protected ConfigurePytestInfoBar(IServiceProvider site) : base(site) {
+            _interpreterOptionsService = site.GetComponentModel().GetService<IInterpreterOptionsService>();
+        }
+
+        protected async Task CheckAsync(
+            IPythonInterpreterFactory interpreterFactory,
+            TestFrameworkType currentTestingFramework,
+            bool suppressPytestConfigInfoBar,
+            string pyTestConfigFilePath,
+            string caption,
+            string context
+        ) {
+            if (IsCreated ||
+                IsGloballySuppressed ||
+                currentTestingFramework == TestFrameworkType.Pytest ||
+                string.IsNullOrEmpty(pyTestConfigFilePath) ||
+                suppressPytestConfigInfoBar ||
+                interpreterFactory == null
+            ) {
+                return;
+            }
+            _interpreterFactory = interpreterFactory;
+            InfoBarContext = context;
+
+
+            //Simplify syntax TODO
+            var packageManagers = _interpreterOptionsService.GetPackageManagers(_interpreterFactory);
+            var getPackagesTask = new List<Task<IList<PackageSpec>>>();
+            foreach (var packageManager in packageManagers) {
+                getPackagesTask.Add(packageManager.GetInstalledPackagesAsync(CancellationToken.None));
+            }
+
```

Comment:
minor suggestion. could use linq to make task a one liner.
 var tasks = packageManagers.Select(p => p.GetInstalledPackagesAsync(  ));
    var results = await Task.WhenAll(tasks);